crates/legacy/src/service.rs

```diff
@@ -1089,16 +1419,56 @@ pub async fn run_permissioned_standalone_builder_service<
     }
 }
 
-/*
-Utility functions to handle the hotshot events
-*/
+/// [HandleDaEventError] represents the internal class of errors that can
+/// occur when attempting to process an incoming da proposal event.  More
+/// specifically these are the class of error that can be returned from
+/// [handle_da_event_implementation].
+#[derive(Debug)]
+enum HandleDaEventError<Types: NodeType> {
+    SenderIsNotLeader,
+    SignatureValidationFailed,
+    BroadcastFailed(async_broadcast::SendError<MessageType<Types>>),
```

Comment:
We could probably do this.  Though if we introduce a new error edge case it might occur in one of the functions but not the other.

I suppose the question I have is:  Is this a desired behavior?  Or is it better to have a separate `enum` of `variant` errors per method, even if many or all of the cases are duplicated?

Reply:
Generally, I'd prefer separate enums, but here the quorum and DA proposal handling are very similar, and even if one has an error case that the other doesn't, we could have a `HandleProposalEventError::QuorumProposalTooLarge` (just a made-up example) variant to make it explicit that it's for the quorum proposal.

However, I don't have a strong opinion. The current way might make debugging easier. I think it's also fine to leave it as is.